src/RustyObjectStore.jl

```diff
@@ -81,52 +98,203 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
+        end
+
+        return new(params)
+    end
+end
+
+abstract type AsConfig end
+
+function as_config(wrapper::AsConfig) end
+
 """
     $TYPEDEF
 
 # Arguments
-$TYPEDFIELDS
+- `url::String`: Url of the object store container root path.
+  It must include the cloud specific url scheme (s3://, azure:// ...).
+- `params::Dict{String, String}`: A set of key-value pairs to configure access to the object store.
+  Refer to the object_store crate documentation for the list of all supported parameters.
 """
-struct AzureCredentials
-    "Azure account name"
-    account::String
-    "Azure container name"
-    container::String
-    "Azure access key"
-    key::String
-    "(Optional) Alternative Azure host. For example, if using Azurite."
-    host::String
-    function AzureCredentials(account::String, container::String, key::String, host::String="")
-        return new(account, container, key, host)
+struct Config <: AsConfig
+    config_string::String
+    function Config(url::String, params::Dict{String, String})
+        dict = merge(Dict("url" => url), params)
+        config_string = JSON3.write(dict)
+        new(config_string)
     end
 end
-function Base.show(io::IO, credentials::AzureCredentials)
-    print(io, "AzureCredentials("),
-    print(io, repr(credentials.account), )
-    print(io, ", ", repr(credentials.container))
-    print(io, ", ", "\"*****\"") # don't print the secret key
-    !isempty(credentials.host) && print(io, ", ", repr(credentials.host))
+
+as_config(wrapper::Config) = wrapper
+
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `storage_account_name::String`: Azure storage account name.
+- `container_name::String`: Azure container name.
+- `storage_account_key::Option{String}`: (Optional) Azure storage account key.
+- `storage_sas_token::Option{String}`: (Optional) Azure storage SAS token.
+- `host::Option{String}`: (Optional) Alternative Azure host. For example, if using Azurite.
+- `opts::ClientOptions`: (Optional) Client configuration options.
+"""
+struct AzureConfig <: AsConfig
+    config::Config
+    function AzureConfig(;
+        storage_account_name::String,
+        container_name::String,
+        storage_account_key::Option{String} = nothing,
+        storage_sas_token::Option{String} = nothing,
+        host::Option{String} = nothing,
+        opts::ClientOptions = ClientOptions()
+    )
+        (
+            !isnothing(storage_account_key)
+            && !isnothing(storage_sas_token)
+        ) && error("Should provide either a storage_account_key or a storage_sas_token")
+
+        params = copy(opts.params)
+
+        params["azure_storage_account_name"] = storage_account_name
+        params["azure_container_name"] = container_name
+
+        if !isnothing(storage_account_key)
+            params["azure_storage_account_key"] = storage_account_key
+        end
+
+        if !isnothing(storage_sas_token)
+            params["azure_storage_sas_token"] = storage_sas_token
```

Comment:
is it fine to pass _both_ a `storage_account_key` and a `storage_sas_token`?

Reply:
Will fix!